### PR TITLE
feat: [PL-38840]: Typescript and Jest Upgrades

### PR DESCRIPTION
**Pull request by bhavya-sinha**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [x] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx [hunk 1]
```diff
@@ -42,9 +49,13 @@ const doConfigureOptionsTesting = async (cogModal: HTMLElement): Promise<void> =
   // Click on Allowed Values radio button and then on Select input
   await waitFor(() => expect(getElementByText(cogModal, 'allowedValues')).toBeInTheDocument())
   const allowedValuesRadio = getElementByText(cogModal, 'allowedValues')
-  userEvent.click(allowedValuesRadio)
+  await act(async () => {
+    await userEvent.click(allowedValuesRadio)
+  })
```
Comment (rameshx) [inline src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx:54]: As we are using the latest version of `userEvent`, the only change required would be to replace all `userEvent.x` to `await userEvent.x`. It should not be wrapped in `act`.

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx:54]: Yes, I saw this only after adding `act` to some files. Rest of the files `act` was not added. I am reverting the places I added `act`.

**Review by vikrant-harness — DISMISSED** [review]

**Review by udayvunnam — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

File: src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx [hunk 2]
```diff
@@ -42,9 +42,9 @@ const doConfigureOptionsTesting = async (cogModal: HTMLElement): Promise<void> =
   // Click on Allowed Values radio button and then on Select input
   await waitFor(() => expect(getElementByText(cogModal, 'allowedValues')).toBeInTheDocument())
   const allowedValuesRadio = getElementByText(cogModal, 'allowedValues')
-  userEvent.click(allowedValuesRadio)
+  await userEvent.click(allowedValuesRadio)
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx:45]: await should not be necessary in such cases? Can we not do without it?

It makes sense for findByText or when you want to wait for the element that is to be clicked.

Comment (praneshg239) [reply src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx:45]: It is mandate in the latest version of jest.

Comment (praneshg239) [reply src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx:45]: As it is promise based, if we are not adding `await` then tests are failing.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

**Review by jyotiarora-harness — APPROVED** [review]

File: src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx (hunk 1 above)
Comment (bhavya-sinha) [reply src/modules/10-common/components/ConfigureOptions/SelectConfigureOptions/__tests__/SelectConfigureOptions.test.tsx]: fixed all

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**